template.rb

```diff
@@ -135,6 +135,111 @@
 HEREDOC
 create_file 'bin/update', BIN_UPDATE, force: true
 
+BIN_BUILD = <<~HEREDOC.strip_heredoc
+  #!/usr/bin/env bash
+
+  set -o errexit
+  set -o pipefail
+  set -o nounset
+
+  echo "=========== setting env variables ==========="
+  export RAILS_ENV='test'
+  export BUNDLE_APP_CONFIG='.bundle/ci-build'
+
+  echo "=========== install bundler ==========="
+  bundler_version=`tail -n 1 Gemfile.lock | xargs`
+  time gem install bundler:$bundler_version --no-document
+
+  echo "=========== bundle install ==========="
+  time bundle install
+
+  echo "=========== bundle audit ==========="
+  time bundle exec bundle-audit check --update
+
+  echo "=========== zeitwerk check ==========="
+  time bundle exec rails zeitwerk:check
+
+  echo "=========== brakeman ==========="
+  time bundle exec brakeman
+
+  echo "=========== rubocop  ==========="
+  time bundle exec rubocop .
+
+  echo "=========== tests  ============="
+  echo "=========== secrets pull ============="
+  time bundle exec secrets pull -e development -y
+
+  echo "=========== rails db:test:prepare ==========="
+  time bundle exec rails db:test:prepare
+
+  echo "=========== rspec ==========="
+  time bundle exec rspec
```

Comment:
```suggestion
  time bundle exec rspec --format failures
```
Does not output dots but it does print failures. It is a bit faster than the default dot progress (seemed like 10% faster locally for me)

Reply:
Ugh, something I just realised. This format does not echo the seed so there is a flaky failure it will be next to impossible to debug. I guess we need the dots formatter.

Reply:
How often do we need seed? 

Maybe we can find an external formatter that servers our purpose? or write our own? :)

Reply:
We need need it on Loop for example because theres parallel workers there and the exact seed + order matters. I don't think I've ever needed the exact CI seed on a non-Loop project.

I'm curious about the performance improvement, I wasn't able to get the same numbers locally because there was too much variance. @JureCindro was this done on a per spec basis or for the whole suite? I'd love a 10% reduction of test execution time for nearly free 🙈

Reply:
We can always do `puts config.seed` inside `rails_helper.rb` :)

But i will leave this for another time